### PR TITLE
cpack: Apply existing Linux package conventions

### PR DESCRIPTION
**Pull request by theopolis**
This is the first part of checking package consistencies. The goal is to make upgrading from version 3.x to 4.x as simple as possible. I am trying to document the inconsistencies I found in https://github.com/osquery/osquery/issues/5769.

### CODE REVIEW

**Comment by Smjert** [comment]
I sort of done a review in the issue #5769.

File: cmake/packaging.cmake [hunk 1]
```diff
@@ -238,6 +238,8 @@ list(GET OSQUERY_VERSION_COMPONENTS 1 CPACK_PACKAGE_VERSION_MINOR)
 list(GET OSQUERY_VERSION_COMPONENTS 2 CPACK_PACKAGE_VERSION_PATCH)
 
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "osquery is an operating system instrumentation toolchain.")
+set(CPACK_PACKAGE_NAME "osquery")
+set(CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
```
Comment (Smjert) [inline cmake/packaging.cmake]: Any reason why these are explicitly specified? That's the default.

Comment (theopolis) [reply cmake/packaging.cmake]: I found that I had to explicitly set these for the line below: https://github.com/osquery/osquery/pull/5770/files#diff-bfc91d448f2fbd980743d90de3646bbbR259

File: cmake/packaging.cmake [hunk 2]
```diff
@@ -250,12 +252,23 @@ configure_file(cmake/CPackConfig.cmake.in package/CPackConfig.cmake @ONLY)
 set(CPACK_GENERATOR "${PACKAGING_SYSTEM}")
 
 if(DEFINED PLATFORM_LINUX)
+  set(CPACK_PACKAGING_INSTALL_PREFIX "/")
+  set(CPACK_STRIP_FILES TRUE)
+
+  if(CPACK_GENERATOR STREQUAL "TGZ")
+    set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}_1.linux.x86_64")
```
Comment (theopolis) [inline cmake/packaging.cmake]: (referenced from above)

This is why `CPACK_PACKAGE_NAME` and `CPACK_PACKAGE_VERSION` are set explicitly. Without setting those above this line fills in blank values for those variables.

File: cmake/packaging.cmake [hunk 3]
```diff
@@ -250,12 +252,23 @@ configure_file(cmake/CPackConfig.cmake.in package/CPackConfig.cmake @ONLY)
 set(CPACK_GENERATOR "${PACKAGING_SYSTEM}")
 
 if(DEFINED PLATFORM_LINUX)
+  set(CPACK_PACKAGING_INSTALL_PREFIX "/")
```
Comment (theopolis) [inline cmake/packaging.cmake]: Head ups, if you look at the above changes the `/etc/default` absolute values are changed to relative values `etc/default`. I can remove this line `set(CPACK_PACKAGING_INSTALL_PREFIX "/")` and restore the absolute values if that is best?

Comment (theopolis) [reply cmake/packaging.cmake]: I added this line mostly to solve for TGZ building. With the absolute values the build fails as it tries to install to those absolute locations during `make package`. I can try to implement your suggestion of DESTDIR settings later today.

File: cmake/packaging.cmake (hunk 2 above)
Comment (Smjert) [reply cmake/packaging.cmake]: Ah I see, right, I totally missed that.
Those variables are set after including the CPack cmake file, so if you use the before that, they'll be empty.
It's fine, if we keep CPack in the future we will probably move the logic after CPack is included and will be run each time CPack is called (like for what's happening right now with the CPackConfig.cmake.in file).

File: cmake/packaging.cmake (hunk 3 above)
Comment (Smjert) [reply cmake/packaging.cmake]: > Head ups, if you look at the above changes the `/etc/default` absolute values are changed to relative values `etc/default`. I can remove this line `set(CPACK_PACKAGING_INSTALL_PREFIX "/")` and restore the absolute values if that is best?

Yes, this is idiomatic in CMake in general, you use CMAKE_INSTALL_PREFIX to set the prefix to install, which is `/usr` by default, and always use relative paths if you want the prefix to be applied; If not you use absolute paths.
Then If you really have to relocate everything, you set the DESTDIR env variable when you install, then that value will be the root; relative values will still have a their prefix.
This is what CPack internally uses to install the project in its own folder, which later processes to create the package.

 > I added this line mostly to solve for TGZ building. With the absolute values the build fails as it tries to install to those absolute locations during make package. I can try to implement your suggestion of DESTDIR settings later today.

CPACK_SET_DESTDIR is definitely the solution there, you just need to set it ON.

**Comment by theopolis** [comment]
@Smjert, let me know if I followed your advice correctly.

I still would like to test on macOS later tonight and ask that someone test Windows. I'd also like to resolve the expectation of including a default certs.pem in another PR that adds the file and modifies CPack logic to include it in the packages. The goal is to remain consistent with the file structure and features provided compared to 3.x.

**Pull request merged**